carmine-api-airdrop/src/merkle_tree.rs

```diff
@@ -0,0 +1,160 @@
+use serde::Deserialize;
+use starknet_crypto::{pedersen_hash, FieldElement};
+use std::fs::File;
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct Airdrop {
+    pub address: String,
+    amount: u32,
+}
+
+#[allow(dead_code)]
+pub struct MerkleTree {
+    pub root: String,
+    tree: Vec<Vec<Node>>,
```

Comment:
Do I understand this correctly as basically array of arrays of Nodes? No idea how complex it would be to clean up afterwards in Rust since there's no GC, but doesn't it make sense to have the Nodes reference each other? Same with the definition of Node, you store Left and Right as Strings instead of pointers/references to other Nodes... That's the way I would expect a tree data structure to be represented in a modern programming language

This is more of a matter of good software engineering / using more idiomatic code and improving code readability than that it would make it impossible to compute anything.

Reply:
Reworked 🔧